packages/@aws-cdk/aws-stepfunctions/test/state-machine-resources.test.ts

```diff
@@ -239,6 +422,95 @@ describe('State Machine Resources', () => {
     });
   }),
 
+  test('Imported state machine can grant task response access to a role', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const stateMachineArn = 'arn:aws:states:::my-state-machine';
+    const stateMachine = stepfunctions.StateMachine.fromStateMachineArn(stack, 'StateMachine', stateMachineArn);
+    const role = new iam.Role(stack, 'Role', {
+      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+    });
+
+    // WHEN
+    stateMachine.grantTaskResponse(role);
+
+    // THEN
+    expect(stack).toHaveResource('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Statement: [
+          {
+            Action: [
+              'states:SendTaskSuccess',
+              'states:SendTaskFailure',
+              'states:SendTaskHeartbeat',
+            ],
+            Effect: 'Allow',
+            Resource: '*',
+          },
+        ],
+        Version: '2012-10-17',
+      },
+      PolicyName: 'RoleDefaultPolicy5FFB7DAB',
+      Roles: [
+        {
+          Ref: 'Role1ABCC5F0',
+        },
+      ],
+    });
+  }),
+
+  test('Imported state machine can grant access to a role', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const stateMachineArn = 'arn:aws:states:::my-state-machine';
+    const stateMachine = stepfunctions.StateMachine.fromStateMachineArn(stack, 'StateMachine', stateMachineArn);
+    const role = new iam.Role(stack, 'Role', {
+      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+    });
+
+    // WHEN
+    stateMachine.grant(role, ['states:DescribeExecution'], `${stateMachine.executionArn}:*`);
+
+    // THEN
+    expect(stack).toHaveResource('AWS::IAM::Policy', {
+      PolicyDocument: {
```

Comment:
Check this beauty out: https://github.com/aws/aws-cdk/pull/8444, it makes matcher code nicer.